### PR TITLE
Add support for LLVM 12

### PR DESCRIPTION
**Pull request by nathanchance**
Closes: https://github.com/ClangBuiltLinux/continuous-integration2/issues/62

### CODE REVIEW

File: generator.yml [hunk 1]
```diff
@@ -297,6 +297,135 @@ builds:
...
+  - {<< : *arm64,         << : *next,             << : *llvm_full,       boot: true,  llvm_version: *llvm_11}
+  - {<< : *arm64_allmod,  << : *next,             << : *llvm_full,       boot: false, llvm_version: *llvm_11}
+  - {<< : *arm64_allno,   << : *next,             << : *llvm_full,       boot: false, llvm_version: *llvm_11}
+  - {<< : *arm64_allyes,  << : *next,             << : *llvm_full,       boot: false, llvm_version: *llvm_11}
+  - {<< : *i386,          << : *next,             << : *llvm_full,       boot: true,  llvm_version: *llvm_11}
+  - {<< : *mips,          << : *next,             << : *mips_llvm,       boot: true,  llvm_version: *llvm_11}
+  - {<< : *mipsel,        << : *next,             << : *llvm,            boot: true,  llvm_version: *llvm_11}
+  # ppc32 and ppc64: Build disabled (https://github.com/ClangBuiltLinux/continuous-integration2/issues/2)
+  # - {<< : *ppc32,         << : *next,             << : *llvm,            boot: true,  llvm_version: *llvm_11}
+  # - {<< : *ppc64,         << : *next,             << : *ppc64_llvm,      boot: true,  llvm_version: *llvm_11}
```
Comment (nickdesaulniers) [inline generator.yml:392]: the can get turned on now, right?

File: generator.yml [hunk 2]
```diff
@@ -286,9 +287,8 @@ builds:
   - {<< : *i386,          << : *next,             << : *llvm_full,       boot: true,  llvm_version: *llvm_latest}
   - {<< : *mips,          << : *next,             << : *mips_llvm,       boot: true,  llvm_version: *llvm_latest}
   - {<< : *mipsel,        << : *next,             << : *llvm,            boot: true,  llvm_version: *llvm_latest}
-  # ppc32 and ppc64: Build disabled (https://github.com/ClangBuiltLinux/continuous-integration2/issues/2)
-  # - {<< : *ppc32,         << : *next,             << : *llvm,            boot: true,  llvm_version: *llvm_latest}
-  # - {<< : *ppc64,         << : *next,             << : *ppc64_llvm,      boot: true,  llvm_version: *llvm_latest}
+  - {<< : *ppc32,         << : *next,             << : *llvm,            boot: true,  llvm_version: *llvm_latest}
+  - {<< : *ppc64,         << : *next,             << : *ppc64_llvm,      boot: true,  llvm_version: *llvm_latest}
```
Comment (nickdesaulniers) [inline generator.yml:291]: it looks the ppc builds are re-enabled (separate commit, plz) for next, but not the other trees. Intentional?

File: generator.yml [hunk 3]
```diff
@@ -297,6 +297,135 @@ builds:
...
+  - {<< : *arm64,         << : *mainline,         << : *llvm_full,       boot: true,  llvm_version: *llvm_11}
+  - {<< : *arm64_allmod,  << : *mainline,         << : *llvm_full,       boot: false, llvm_version: *llvm_11}
+  - {<< : *arm64_allno,   << : *mainline,         << : *llvm_full,       boot: false, llvm_version: *llvm_11}
+  - {<< : *arm64_allyes,  << : *mainline,         << : *llvm_full,       boot: false, llvm_version: *llvm_11}
+  - {<< : *i386,          << : *mainline,         << : *llvm_full,       boot: true,  llvm_version: *llvm_11}
+  - {<< : *mips,          << : *mainline,         << : *mips_llvm,       boot: true,  llvm_version: *llvm_11}
+  - {<< : *mipsel,        << : *mainline,         << : *llvm,            boot: true,  llvm_version: *llvm_11}
+  # ppc32 and ppc64: Build disabled (https://github.com/ClangBuiltLinux/continuous-integration2/issues/2)
+  # - {<< : *ppc32,         << : *mainline,         << : *llvm,            boot: true,  llvm_version: *llvm_11}
+  # - {<< : *ppc64,         << : *mainline,         << : *ppc64_llvm,      boot: true,  llvm_version: *llvm_11}
```
Comment (nickdesaulniers) [inline generator.yml:365]: ppc

**Review by nickdesaulniers — CHANGES_REQUESTED** [review]
I think just some ppc builds should be enabled, otherwise LGTM

File: generator.yml (hunk 1 above)
Comment (nathanchance) [reply generator.yml:392]: No, LLVM 11 does not have that bug fixed.

File: generator.yml (hunk 2 above)
Comment (nathanchance) [reply generator.yml:291]: No, these are not re-enabled technically, these are being copied from llvm_tot, where they are enabled, The diff is weird.

File: generator.yml (hunk 1 above)
Comment (nickdesaulniers) [reply generator.yml:392]: Is https://github.com/ClangBuiltLinux/linux/issues/1244 the issue that should be linked in the comment, then?

Comment (nathanchance) [reply generator.yml:392]: I can update it if you would like. That issue is linked in the Linaro issue that is linked in the continuous-integration issue.

Comment (nickdesaulniers) [reply generator.yml:392]: Even linking to the linaro issue would be one less level of indirection.  Sounds like tuxmake needs to permit setting  CROSS_COMPILE?

Comment (nathanchance) [reply generator.yml:392]: > Even linking to the linaro issue would be one less level of indirection.

Sure, done.

> Sounds like tuxmake needs to permit setting CROSS_COMPILE?

Yes or going to a more neutral triple like `powerpc-linux-gnu-`.

**Review by nickdesaulniers — APPROVED** [review]
LGTM thanks @nathanchance 🍹 🍍

**Comment by nathanchance** [comment]
Thank you for the review as always!

**Pull request merged**